trunner/runners/ARMV7M7Runner.py

```diff
@@ -0,0 +1,161 @@
+
+#
+# Phoenix-RTOS test runner
+#
+# ARMV7M7 runner
+#
+# Copyright 2021 Phoenix SYstems
+# Authors: Jakub Sarzyński, Damian Loewnau
+#
+
+import time
+import sys
+import signal
+
+from .common import Psu, Phoenixd, PhoenixdError, PloError, PloTalker, Runner, DeviceRunner
+from .common import GPIO, logging, phd_error_msg, rootfs
+from trunner.tools.color import Color
+from pexpect.exceptions import TIMEOUT, EOF
+from abc import ABC, abstractmethod
+
+
+def handler(signum, frame):
+    logging.error("\nThe boot mode's change wasn't validated by pressing enter!\n")
+    sys.exit(1)
```

Comment:
I don't like exiting from trunner in some error and potentially leaving test setup in invalid state (but that is a generic problem I've pointed out in @niewim19 PR recently).

Timeout should result in exception, thus failed testcase (or whole testing depending on when it happened) and should be propagated to the upper layer.

Also: using signals might not be the best way to set timeout for user input - if we would ever think of making this toolkit multi-threaded. I'm not focusing on reset-by-human use case as it's not the main point of this PR

Reply:
I've tested a few solutions and the last one is my proposal.

If host_pc reboot fail is when flashing, whole testing is stopped, but when loading tests, the specific test case fails.
![Screenshot from 2021-11-26 18-09-33](https://user-images.githubusercontent.com/77304431/143613738-98b7656a-4a91-4de2-ae66-c384906296c7.png)
![Screenshot from 2021-11-26 18-10-23](https://user-images.githubusercontent.com/77304431/143613745-6d7b68c4-3b61-4d93-bb8c-fe385642e838.png)